### ISSUE TITLE
Deployment on Vercel fails with fresh Blitz App

### ISSUE THREAD

**Issue by rfermann** [blitz-js/blitz#2125]
### What is the problem?

The deployment on Vercel fails after creating and deploying a fresh Blitz App. The deployed page first loads and then errors out. Page can be visited [here]:(https://vercel-deployment-a1lcubdas-rfermann.vercel.app/ ), the github repo can be found [here]:(https://github.com/rfermann/vercel-deployment).

The same setup is working fine on local machine.

### Paste all your error logs here:

Vercel Logs show the following error:
```
[POST] /api/users/queries/getCurrentUser
09:17:32:80
info  - Loaded env from /var/task/.env
2021-03-18T08:17:34.060Z	undefined	ERROR	Uncaught Exception 	{"errorType":"Error","errorMessage":"Could not find a production build in the '/var/task/.next' directory. Try building your app with 'next build' before starting the production server. https://err.sh/vercel/next.js/production-start-no-build-id","stack":["Error: Could not find a production build in the '/var/task/.next' directory. Try building your app with 'next build' before starting the production server. https://err.sh/vercel/next.js/production-start-no-build-id","    at Server.readBuildId (/var/task/node_modules/next/dist/next-server/server/next-server.js:148:355)","    at new Server (/var/task/node_modules/next/dist/next-server/server/next-server.js:3:120)","    at Object.<anonymous> (/var/task/___next_launcher.js:16:20)","    at Module._compile (internal/modules/cjs/loader.js:1063:30)","    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)","    at Module.load (internal/modules/cjs/loader.js:928:32)","    at Function.Module._load (internal/modules/cjs/loader.js:769:14)","    at Module.require (internal/modules/cjs/loader.js:952:19)","    at require (internal/modules/cjs/helpers.js:88:18)","    at _tryRequire (/var/runtime/UserFunction.js:75:12)"]}
Unknown application error occurred
Error
```

### Paste all relevant code snippets here:

No custom code snippets used, just a barebones Blitz App without any changes.

### What are detailed steps to reproduce this?

1. run blitz new someApp in your terminal
2. follow https://blitzjs.com/docs/deploy-vercel to prepare deployment

Only difference to the description is that I used a SQL database on a different provider (GCP) without a load balancer.

### Run `blitz -v` and paste the output here:

```
macOS Big Sur | darwin-x64 | Node: v15.8.0

blitz: 0.33.0 (global)
blitz: 0.33.0 (local)

  Package manager: yarn
  System:
    OS: macOS 11.2.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 26.84 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.8.0 - ~/.nvm/versions/node/v15.8.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v15.8.0/bin/yarn
    npm: 7.5.2 - ~/.nvm/versions/node/v15.8.0/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: ~2.17 => 2.17.0
    blitz: 0.33.0 => 0.33.0
    prisma: ~2.17 => 2.17.0
    react: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0
    react-dom: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0
    typescript: ~4.1 => 4.1.5
```

**Comment by flybayer** [comment]
Ok that's really weird. I would try again today, maybe it's a temporary issue.

If still not working, contact Vercel support.

**Comment by Skn0tt** [comment]
I had a similar problem with quirrel.dev. It was fixed by rolling back to 0.32.1. Judging by the diff between 0.33.0 and 0.32.1, I can't find a reason why this should happen.

**Comment by RobertBroersma** [comment]
I too experienced this exact problem after upgrading to 0.33.0. Downgrading to 0.32.1 fixed it as @Skn0tt mentioned.

**Comment by JonasBa** [comment]
@flybayer I think this is duplicate of the one I just opened (sorry, I haven't seen it through search) - https://github.com/blitz-js/blitz/issues/2132

I attached the detailed stack trace, so feel free to close/move the discussion to either of the two

**Comment by rfermann** [comment]
@JonasBa it might still be 2 different issues. My actual deployment passed without issues. But all the pages using the api are erroring out.

**Comment by flybayer** [comment]
This almost certainly caused by Vercel or something they changed. I have contacted them for help

**Comment by ofhouse** [comment]
Some info: Vercel uses `target: 'server'` instead of `target: 'experimental-serverless-trace` for versions of Next.js 10.0.9 and higher on their build system.

**Comment by flybayer** [comment]
@ofhouse whaaaat, why did they change that?

I found this comment about them removing the docs for the target option which could provide some insight: https://github.com/vercel/next.js/issues/20790#issuecomment-755178273

> - Significantly slower compilation (as all deps are bundled in the page bundles, there's better ways to handle it nowadays, especially with AWS lambdas)
> - Lambas no longer have cold start problems related to zip size, only related to code loading, more code loaded takes longer to boot, Next.js next start by default only loads code needed for the incoming request.

**Comment by waptik** [comment]
Seems like i'm not the only one with this issue.
Deployment was working perfectly with blitz v0.32.1 but now gets 500 error with v0.33.0.
I downgraded back to v0.32.1 per @Skn0tt suggestion and it now works.
Thanks man!!!

**Comment by leerob** [comment]
Hey everyone! We've been investigating this on the Vercel side. First, we encountered a TypeScript issue with the official Blitz + Next example. Should I create an issue to update this? Secondly, we ran into a module not found error.
![image](https://user-images.githubusercontent.com/9113740/112666915-ac92bb00-8e2a-11eb-9235-ebc86f5d9606.png)

We're still investigating on our side and working with @flybayer 😄

**Comment by flybayer** [comment]
@leerob we already have an issue for that: https://github.com/blitz-js/blitz/issues/2132

**Comment by flybayer** [comment]
Vercel says this should be fixed now. Can someone test?

**Comment by waptik** [comment]
> Vercel says this should be fixed now. Can someone test?

I can confirm it's been fixed, at least at my end.
I no longer get error 500.
I just upgraded blitz to latest version

**Comment by sakulstra** [comment]
> Vercel says this should be fixed now. Can someone test?

works for me as well

**Comment by leerob** [comment]
Glad it’s working for everyone 👍🏼 Thanks for bearing with us 🙏🏼

**Comment by rfermann** [comment]
Same here, new deployment with the same code base is working. Thanks for fixing.

**Comment by brapifra** [comment]
It's still happening to me with `blitz@0.34.1`. Anybody else?
